packages/sdk-effects/storage/src/lib/storage.effects.ts

```diff
@@ -6,9 +6,21 @@
  */
 import { CustomStorageObject } from '@forgerock/sdk-types';
 
+export interface StorageClient<Value> {
+  get: () => Promise<Value | GenericError | null>;
+  set: (value: Value) => Promise<void | {
+    code: string;
+    message: string;
+    type: string;
+  }>;
+  remove: () => Promise<void>;
+}
+
 export interface StorageConfig {
-  storeType: CustomStorageObject | 'localStorage' | 'sessionStorage';
+  type: 'custom' | 'localStorage' | 'sessionStorage';
   prefix?: string;
+  name: string;
```

Comment:
Just curious if there was a challenge you came across that resulted in the addition of the `name` property on storage. Does it really need to be required?

Reply:
No challenge per se, but the arguments seemed a bit odd when split out like they were. I just moved the additional parameters into this single config object so that now there's only one parameter. There requirements of each remain the same.